tests/pytest/test_binary_value.py

```diff
@@ -8,6 +8,36 @@
 TRUNCATION_MATCH = r"\d+-bit value requested, truncating value"
 
 
+def test_init_zero_length():
+    with pytest.warns(RuntimeWarning, match=TRUNCATION_MATCH):
+        bin1 = BinaryValue(value=0, n_bits=0, binaryRepresentation=BinaryRepresentation.UNSIGNED)
```

Comment:
Why does this warn?

Reply:
As x is a integer at the beginning, I think we should test for 0 before calling bin(). This should be faster. Please have a look at 
e359032fef8abcc33eee14ae2bc6e02df34a7c94

Reply:
Probably we need to do something similar with _convert_to_twos_comp and  _convert_to_signed_mag.

Reply:
I had also to adjust _convert_to_twos_comp / _convert_to_signed_mag